app/src/main/java/ch/epfl/favo/viewmodel/FavorViewModel.java

```diff
@@ -95,39 +93,60 @@ private CompletableFuture updateFavorForCurrentUser(
         .thenCompose(o -> getFavorRepository().updateFavor(favor));
   }
 
+  public CompletableFuture commitFavor(Favor favor, int change) {
+    Favor tempFavor = (Favor) favor.clone();
+    return changeUserActiveFavorCount(
+            DependencyFactory.getCurrentFirebaseUser().getUid(),
+            false,
+            change) // if user can accept favor then post it in the favor collection
+            .thenCompose((f) -> getFavorRepository().updateFavor(tempFavor));
+  }
+
   // save address to firebase
   @Override
   public CompletableFuture requestFavor(Favor favor) {
```

Comment:
if the change variable can only ever have a 1 or 0 value we should probably change it to use a bool to be more explicit on its use.

Reply:
It will not always be just 1 or -1. When we batch-delete in the list view it might be larger